### PR TITLE
Refactor: Remove Avatar platform specific code

### PR DESCRIPTION
**Pull request by kidroca**
### Details

The avatar component uses a workaround to prevent a flicker bug in react-native-web
We submitted a fix for react-native-web and the workaround is no longer needed

#### Why was the Image component updated
The code contributed to the flicker because the `imageSource` starts with `undefined` and is only updated after mount - this causes the first render pass to show no image (this happens very quickly, but sometimes a flicker can be seen)

### Fixed Issues

$ https://github.com/Expensify/App/issues/14278   
PROPOSAL: https://github.com/Expensify/App/issues/9183#issuecomment-1381676114
FORK PR: https://github.com/Expensify/react-native-web/pull/15

#### There should be no flicker
1. Log into an account on New Expensify
2. Navigate to a chat where you were the last person to send a message
3. Send a message in that chat
4. Confirm that the avatar on the left hand side of the message that you just sent does not flickers.

#### Image `onLoad` was slightly changed, it should not cause a regression
- Thumbnails should work as before / resize correctly after load
- Attachment preview modals should work as before

### Offline tests

- Sending a message while offline should not case the avatar to flicker

### QA Steps

1. Log into an account on New Expensify
2. Navigate to a chat where you were the last person to send a message
3. Send a message in that chat
4. Confirm that the avatar on the left hand side of the message that you just sent does not flickers.

- Thumbnails should work as before / resize correctly after load
- Attachment preview modals should work as before

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/12156624/222922567-1da327e5-93df-4c33-acb2-fb55eb14acd4.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/12156624/223125487-808e4ae4-c84b-4503-8138-48611473b1c0.mp4

</details>

<details>
<summary>Mobile Web - Safari</summary>

Screenshot 2023-03-04 at 20 46 00

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/12156624/222923288-789d33fa-0334-4cf9-822d-0cbe29fa7500.mov

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/12156624/222923523-6ff1ec89-6ccc-4ccc-9c39-47aaec32674a.mp4

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/12156624/223123941-9df60a91-0525-4859-92d6-5211cbe8fd1b.mp4

</details>

### CODE REVIEW

File: package.json
Comment (kidroca) [inline package.json:44]: This is the complied code containing the flicker fix
After it's merged the package should be updated to point to the new version
- https://github.com/Expensify/react-native-web/pull/15

File: src/components/Avatar.js
Comment (kidroca) [inline src/components/Avatar.js:124]: The `defaultSource` workaround is no longer needed.
We just pass `source`

File: src/components/Image/index.js
Comment (kidroca) [inline src/components/Image/index.js:15]: This would render no image after the first render and only after add the `imageSource`
We don't have to use state for `imageSource` - it doesn't save us from re-renders, because most usages pass source like `<Image soruce={{ uri: '...' }} />` (a new object created inline)

**Review by kidroca — COMMENTED** [review]
Added some notes to aid the reviewer

File: src/components/Image/index.js
Comment (mollfpr) [inline src/components/Image/index.js:59]: This is not part of your changes, but can we cleanup this by using `omit` instead? `source` and `onLoad` has not been used.

Comment (kidroca) [reply src/components/Image/index.js]: Done

**Comment by mollfpr** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



https://user-images.githubusercontent.com/25520267/223139156-aa3aa75e-96dc-43c1-8da7-a1fa20191647.mov


https://user-images.githubusercontent.com/25520267/223139193-49f1ba15-0706-45b5-ad5b-7785fbc84b00.mp4


</details>

<details>
<summary>Mobile Web - Chrome</summary>



https://user-images.githubusercontent.com/25520267/223139220-086d40e5-5ea1-4164-a835-8fce5779dcd9.mp4


</details>

<details>
<summary>Mobile Web - Safari</summary>



https://user-images.githubusercontent.com/25520267/223139284-2e47e397-a5a7-404d-947e-d52bf6cd37ad.mp4


https://user-images.githubusercontent.com/25520267/223139246-aad09947-f80f-4a8c-887c-04f432e9c6da.mp4


</details>

<details>
<summary>Desktop</summary>




https://user-images.githubusercontent.com/25520267/223149050-1f03afdf-3d2e-4391-a82d-dd9acec93d8b.mp4


</details>

<details>
<summary>iOS</summary>



https://user-images.githubusercontent.com/25520267/223147320-04a266a4-557f-4bd4-95a9-4b35204fd174.mp4


</details>

<details>
<summary>Android</summary>



https://user-images.githubusercontent.com/25520267/223147350-1e1d8867-aa92-4c18-91e6-7697daeae00a.mp4


https://user-images.githubusercontent.com/25520267/223147447-a500edae-1c40-42c6-b09e-f4a88efec777.mp4


</details>

**Review by mollfpr — DISMISSED** [review]
Great work on this @kidroca! Looks good to me and tests well 👍

**Comment by kidroca** [comment]
Thanks @mollfpr 
I just want to remind that the current PR would have wait and be updated with the new `@expensify/react-native-web` version, once it is merged:
- https://github.com/Expensify/react-native-web/pull/15

BTW no one is assigned to review it

---

Current PR servers to verify the fix is working in App, but it has to wait for the actual fix to be released in `@expensify/react-native-web`

File: package.json
Comment (bondydaa) [reply package.json:44]: should we throw a hold on this PR until https://github.com/Expensify/react-native-web/pull/15 is merged?

**Review by bondydaa — COMMENTED** [review]
👍 looks good, I'll approve once we've gotten the react-native-web PR merged and the package.json updated with the correct version number.

**Review by mountiny — COMMENTED** [review]
I am going to try to release the new version of our fork, @kidroca can you resolve the merge conflicts in the meantime, thanks!

**Comment by kidroca** [comment]
> I am going to try to release the new version of our fork

I don't see the new `@expensify/react-native-web` version, latest is v0.18.12 - published 2 months ago: https://www.npmjs.com/package/@expensify/react-native-web

File: src/components/Image/index.js
Comment (kidroca) [reply src/components/Image/index.js:18]: The `debounce` here was removed because it no longer has any effect (and was a workaround) as discussed here: https://github.com/Expensify/App/pull/9279#discussion_r1140746497

(Also it's inconsistent with the `.native` component where no debounce is used)

There are further changes that we can add to the Image component and I can open a follow up PR:
1. remove `configureOnLoad` - this was a workaround needed because at the time `react-native-web` did not provide dimension in the original `onLoad` callback
2. Turn `Image` into a functional component - no lifecycle and state are used. 

I do both of this changes here:
- https://github.com/Expensify/App/pull/13036

but the PR is on hold

**Comment by mollfpr** [comment]
### Screenshots/Videos
<details>
<summary>Web</summary>



https://user-images.githubusercontent.com/25520267/227570985-7ff944cb-001a-4450-98ae-b3e71e7dcaef.mp4





</details>

<details>
<summary>Mobile Web - Chrome</summary>



https://user-images.githubusercontent.com/25520267/227571146-2699e580-1f98-4ef7-9f68-ddf05233f2fd.mp4


</details>

<details>
<summary>Mobile Web - Safari</summary>



https://user-images.githubusercontent.com/25520267/227571008-da9653dd-c4c5-432a-b82d-2f72ee42b8ff.mp4


</details>

<details>
<summary>Desktop</summary>


https://user-images.githubusercontent.com/25520267/227571116-00d3522b-f1c5-4196-896b-654c65a5e3ec.mp4
</details>

<details>
<summary>iOS</summary>



https://user-images.githubusercontent.com/25520267/227571032-ca534106-230e-4555-94f1-0ad25dd0ed40.mp4


</details>

<details>
<summary>Android</summary>



https://user-images.githubusercontent.com/25520267/227571056-dc7f8475-5e1d-45c2-bbb2-2bf9b4394a5e.mp4


</details>

**Review by mollfpr — APPROVED** [review]
Test well 👍

**Review by mountiny — APPROVED** [review]
Thank you very much, going to go ahead and merge given Bondy has approved this before. Thank for quick work @kidroca and thanks for testing @mollfpr

**Pull request merged**
